lib/matplotlib/figure.py

```diff
@@ -1073,30 +1073,39 @@ def fixlist(args):
         return key
 
     def _process_projection_requirements(
-            self, *args, polar=False, projection=None, **kwargs):
+            self, *args, axes_class=None, polar=False, projection=None,
+            **kwargs):
         """
         Handle the args/kwargs to add_axes/add_subplot/gca, returning::
 
             (axes_proj_class, proj_class_kwargs, proj_stack_key)
 
         which can be used for new axes initialization/identification.
         """
-        if polar:
-            if projection is not None and projection != 'polar':
+        if axes_class is not None:
+            if polar or projection is not None:
                 raise ValueError(
-                    "polar=True, yet projection=%r. "
-                    "Only one of these arguments should be supplied." %
-                    projection)
-            projection = 'polar'
-
-        if isinstance(projection, str) or projection is None:
-            projection_class = projections.get_projection_class(projection)
-        elif hasattr(projection, '_as_mpl_axes'):
-            projection_class, extra_kwargs = projection._as_mpl_axes()
-            kwargs.update(**extra_kwargs)
+                    "Cannot combine 'axes_class' and 'projection' or 'polar'")
+            projection_class = axes_class
```

Comment:
You and I have crossing PRs, so its pretty important for my rebases that you have tests if possible

Reply:
sorry about that, added a test

Reply:
This line still doesn't have a test according to codecov.  I can probably be careful in my rebase, but....

Reply:
hopefully better now